### ISSUE TITLE
MouseEvent type on Legend component doesn't correspond to React MouseEvent

### ISSUE THREAD

**Issue by Rashair** [recharts/recharts#3362]
- [X] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/recharts-issue-template-forked-zesr25?file=/src/index.tsx:482-483)

### Steps to reproduce
Create a function for Legend `onClick` handle with types and all 3 arguments and assign it to the handle.

### What is expected?
No type error.

### What is actually happening?
There's a type error.

| Environment | Info |
|---|---|
| Recharts | v2.4.1 |
| React | 18.2.0 |
| System | Windows 22621.1105 |
| Browser | 109.0.5414.120 |

---
The error doesn't occur with 1 / 2 args, only with MouseEvent.

**Comment by ckifer** [comment]
Definitely some not-ideal typing going on here, but to get around it you can do
`onClick={(data, index, event) => onLegendIconClick(data, index, event)}`

**Comment by ckifer** [comment]
If you look at the types used

https://github.com/recharts/recharts/blob/a532ef6ebc624cd3cab070ba29c516d7f473e2a7/src/component/DefaultLegendContent.tsx#L60

They are the same as you are using in your demo here. Not sure why they aren't equal

**Comment by ckifer** [comment]
Recharts needs to update its React types version

https://github.com/recharts/recharts/blob/master/package.json#L103

@nikolasrieble for vis, this actually quite the issue

**Comment by ckifer** [comment]
I can get this to go away locally by using 

` ev: React.MouseEvent<React.ReactElement, MouseEvent>,`

But I can't get it to go away in sandbox

**Comment by nikolasrieble** [comment]
@ckifer Could you please expand on the issue? Is the React types version out of sync with the React version?

**Comment by ckifer** [comment]
Yes - we use types from React 16.0 which are different than those from React 16.8+. 

In this case React.MouseEvent just behaves incredibly odd, but overall we should keep our types packages in line with the libraries they support.

**Comment by ckifer** [comment]
`MouseEvent<React.ReactNode>` works in the sandbox - that'll unblock you for now @Rashair
